tests/ui/self/arbitrary_self_types_raw_pointer_trait.rs

```diff
@@ -34,12 +32,6 @@ impl Foo for u32 {
 }
 
 fn main() {
-    let null_i32 = ptr::null::<i32>() as *const dyn Foo;
-    let null_u32 = ptr::null::<u32>() as *const dyn Foo;
-
-    assert_eq!("I'm an i32!", null_i32.foo());
-    assert_eq!("I'm a u32!", null_u32.foo());
```

Comment:
Is that really UB? There's only raw pointers involved and they are not dereferenced?

Reply:
```
error: Undefined Behavior: dereferencing pointer failed: null pointer is a dangling pointer (it has no provenance)
  --> tests/ui/self/arbitrary_self_types_raw_pointer_trait.rs:40:31
   |
40 |     assert_eq!("I'm an i32!", null_i32.foo());
   |                               ^^^^^^^^^^^^^^ dereferencing pointer failed: null pointer is a dangling pointer (it has no provenance)
```